MoltenVK/MoltenVK/OS/MTLRenderPipelineColorAttachmentDescriptor+MoltenVK.m

```diff
@@ -0,0 +1,99 @@
+/*
+ * MTLRenderPipelineDescriptor+MoltenVK.m
+ *
+ * Copyright (c) 2018-2023 Chip Davis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "MVKCommonEnvironment.h"
+
+#if MVK_USE_METAL_PRIVATE_API
+
+#include "MTLRenderPipelineDescriptor+MoltenVK.h"
+
+typedef NSUInteger MTLLogicOperation;
+// This is disgusting. We shouldn't have to do this. We can't even rely on this.
+typedef struct MTLRenderPipelineAttachmentDescriptorPrivate {
+	union {
+		struct {
+			BOOL blendingEnabled : 1;
+			MTLBlendOperation rgbBlendOperation : 3;
+			MTLBlendOperation alphaBlendOperation : 3;
+			MTLBlendFactor sourceRGBBlendFactor : 5;
+			MTLBlendFactor sourceAlphaBlendFactor : 5;
+			MTLBlendFactor destinationRGBBlendFactor : 5;
+			MTLBlendFactor destinationAlphaBlendFactor : 5;
+			MTLColorWriteMask writeMask : 4;
+			BOOL logicOpEnabled : 1;
+			MTLLogicOperation logicOp : 4;
+			MTLPixelFormat pixelFormat : 28;
+		};
+		struct {
+			uint64_t bits;
+		};
+	};
+} MTLRenderPipelineAttachmentDescriptorPrivate;
```

Comment:
What's going on here? Is this a private Metal struct, that has bitwise density, and that we've reversed engineered? 

If so, how was that even done? And the obvious concern is that this adds a whole different level of fragility to the builds and runtimes.

Reply:
It was present in the Objective-C metadata for `Metal.framework`. At the time I wrote that, I couldn't see a method for setting this. Frankly, I don't like it either, because it *is* an internal implementation detail of Metal. Now there is such a method, so if you like, I can remove this.

Reply:
Yeah. I respect your deep research, but at least for now, I think I'd feel much better relying on just the private `logicOpEnabled` property (which you use below), and save this level of intrusion for if in future that we get complaints that the private `logicOpEnabled` property doesn't cover enough platforms or OS versions.